File: Ruido.java
import java.util.Random;

class Ruido {
    Random random = new Random();

    public String aplicarRuido(String binario, double tasaError) {
        StringBuilder ruidoso = new StringBuilder();
        for (char bit : binario.toCharArray()) {
            if (random.nextDouble() < tasaError) {
                bit = (bit == '1') ? '0' : '1'; // flip
            }
            ruidoso.append(bit);
        }
        System.out.println("Mensaje con ruido aplicado: " + ruidoso);
        return ruidoso.toString();
    }
}

File: Ruido.py
import random

class Ruido:
    def aplicar_ruido(self, binario: str, tasa_error: float) -> str:
        ruidoso = []
        for bit in binario:
            if random.random() < tasa_error:
                bit = "1" if bit == "0" else "0"
            ruidoso.append(bit)
        ruidoso_str = "".join(ruidoso)
        return ruidoso_str
